### DISCUSSION TITLE
通过bytehook_hook_single hook pthread_create函数，新函数只会被执行一次

### DISCUSSION THREAD

**Discussion by wanggaohui** [bytedance/bhook#54] [Q&A]
### the Feature, Motivation and Pitch

![hook pthread代码](https://user-images.githubusercontent.com/40512643/190364534-ade26213-e8ab-4b99-8266-7e42b92375d7.png)

**Comment by caikelun** [comment]
proxy函数贴一下？

**Reply by wanggaohui** [reply]
![image](https://user-images.githubusercontent.com/40512643/190367364-3fc554b5-ee78-4dd3-a9bc-8bf224e996ec.png)

这是pthread_create的新函数

**Reply by wanggaohui** [reply]
@caikelun  帮忙看看是不是proxy函数不对的原因，还是bytehook_hook_single用法不对

**Reply by wanggaohui** [reply]
![image](https://user-images.githubusercontent.com/40512643/190395054-51fe3de9-9a5d-4264-9aff-548d78b84ec1.png)

原先用xhook这么写是ok的，proxy函数没有修改过，但改成bhook就变成这样了

**Comment by caikelun** [comment]
proxy函数写法不对，请先阅读bytehook的文档吧
https://github.com/bytedance/bhook/tree/main/doc#readme

**Reply by caikelun** [reply]
bytehook有两种模式，在初始化时指定，你在bytehook_init中指定的是AUTOMATIC模式，proxy函数需要用AUTOMATIC模式对应的写法。如果在bytehook_init中指定MANUAL模式，则可以用你目前的proxy函数。
